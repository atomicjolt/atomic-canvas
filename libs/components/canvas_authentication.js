"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.CanvasAuthentication = CanvasAuthentication;
exports["default"] = void 0;

var _lodash = _interopRequireDefault(require("lodash"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _settings = require("atomic-fuel/libs/components/settings");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function CanvasAuthentication(props) {
  var autoSubmit = props.autoSubmit,
      hideButton = props.hideButton,
      buttonText = props.buttonText,
      buttonClassName = props.buttonClassName,
      settings = props.settings,
      overrides = props.overrides;
  var formRef = (0, _react.useRef)(null);
  (0, _react.useEffect)(function () {
    if (autoSubmit) {
      formRef.current.submit();
    }
  }, [autoSubmit]);

  function getButton() {
    if (hideButton) return null;
    return /*#__PURE__*/_react["default"].createElement("input", {
      type: "submit",
      value: buttonText || "Authorize",
      className: buttonClassName
    });
  }

  function renderSettings() {
    var settings = _objectSpread(_objectSpread({}, settings), overrides);

    return _lodash["default"].map(settings, function (value, key) {
      var outValue = value || '';

      if (_lodash["default"].isObjectLike(value)) {
        outValue = JSON.stringify(outValue);
      }

      return /*#__PURE__*/_react["default"].createElement("input", {
        key: key,
        type: "hidden",
        value: outValue,
        name: key
      });
    });
  }

  return /*#__PURE__*/_react["default"].createElement("form", {
    ref: formRef,
    action: settings.canvas_oauth_url,
    method: "post"
  }, getButton(), renderSettings());
}

CanvasAuthentication.defaultProps = {
  overrides: {},
  hideButton: false,
  autoSubmit: false,
  buttonClassName: "",
  buttonText: null
};
CanvasAuthentication.propTypes = {
  overrides: _propTypes["default"].shape({}),
  hideButton: _propTypes["default"].bool,
  autoSubmit: _propTypes["default"].bool,
  settings: _propTypes["default"].shape({
    canvas_oauth_url: _propTypes["default"].string
  }).isRequired,
  buttonClassName: _propTypes["default"].string,
  buttonText: _propTypes["default"].string
};

var _default = (0, _settings.withSettings)(CanvasAuthentication);

exports["default"] = _default;