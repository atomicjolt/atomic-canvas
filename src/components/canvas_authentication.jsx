import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import { withSettings } from 'atomic-fuel/libs/components/settings';

export class CanvasAuthentication extends React.Component {
  static defaultProps = {
    overrides: {},
    hideButton: false,
    autoSubmit: false,
    buttonClassName: "",
    buttonText: null,
  }

  static propTypes = {
    overrides: PropTypes.shape({}),
    hideButton: PropTypes.bool,
    autoSubmit: PropTypes.bool,
    settings: PropTypes.shape({
      canvas_oauth_url: PropTypes.string,
    }).isRequired,
    buttonClassName: PropTypes.string,
    buttonText: PropTypes.string,
  }

  componentDidMount() {
    if (this.props.autoSubmit) {
      this.form.submit();
    }
  }

  getButton() {
    if (this.props.hideButton) return null;
    return <input type="submit" value={this.props.buttonText || "Authorize"} className={this.props.buttonClassName} />;
  }

  renderSettings() {
    const settings = { ...this.props.settings, ...this.props.overrides };
    return _.map(settings, (value, key) => {
      let outValue = value || '';
      if (_.isObjectLike(value)) {
        outValue = JSON.stringify(outValue);
      }
      return (
        <input key={key} type="hidden" value={outValue} name={key} />
      )
    });
  }

  render() {
    return (
      <form
        ref={(ref) => { this.form = ref; }}
        action={this.props.settings.canvas_oauth_url}
      >
        { this.getButton() }
        { this.renderSettings() }
      </form>
    );
  }
}

export default withSettings(CanvasAuthentication);
