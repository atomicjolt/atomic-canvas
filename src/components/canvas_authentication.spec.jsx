import React from 'react';
import { shallow } from 'enzyme';
import { CanvasAuthentication } from './canvas_authentication';

describe('Canvas authentication', () => {
  it('renders a button to authenticate with Canvas', () => {
    const settings = {
      canvas_oauth_url: 'http://www.example.com',
      user:  {
        name: 'doug',
      }
    };
    const result = shallow(<CanvasAuthentication settings={settings} />);
    expect(result).toBeDefined();
  });
});