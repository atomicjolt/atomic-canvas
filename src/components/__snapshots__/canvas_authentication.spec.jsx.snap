// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canvas authentication renders a button to authenticate with Canvas 1`] = `
<form
  action="http://www.example.com"
  method="post"
>
  <input
    className=""
    type="submit"
    value="Authorize"
  />
</form>
`;
